src/mantine-core/src/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -101,27 +101,26 @@ export function SegmentedControl<T extends string = string>({
   const [activePosition, setActivePosition] = useState({ width: 0, translate: 0 });
   const uuid = useUuid(name);
   const refs = useRef<Record<string, HTMLLabelElement>>({});
-  const wrapperRef = useRef<HTMLDivElement>(null);
+  const [ref] = useResizeObserver();
 
   useEffect(() => {
-    const observer = new ResizeObserver(() => {
-      if (_value in refs.current && wrapperRef.current) {
-        const element = refs.current[_value];
-        const rect = element.getBoundingClientRect();
-        setActivePosition({
-          width: rect.width,
-          translate: rect.x - wrapperRef.current.getBoundingClientRect().x - WRAPPER_PADDING,
-        });
-
-        if (!shouldAnimate) {
-          setTimeout(() => {
-            setShouldAnimate(true);
-          }, 4);
-        }
+    if (_value in refs.current && ref.current) {
+      const element = refs.current[_value];
+      const elementRect = element.getBoundingClientRect();
+
+      // Scaled value to undo transforms applied
+      const scaledValue = element.offsetWidth / elementRect.width;
```

Comment:
Please also add a story that will have scale toggle (like in sandbox shared in issue – https://codesandbox.io/s/romantic-breeze-q3e99?file=/src/App.js)